packages/rabbitmq/src/TxSubmitWorker.ts

```diff
@@ -0,0 +1,306 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+import { Channel, Connection, Message, connect } from 'amqplib';
+import { Logger, dummyLogger } from 'ts-log';
+import { ProviderError, ProviderFailure, TxSubmitProvider } from '@cardano-sdk/core';
+import { TX_SUBMISSION_QUEUE } from './rabbitmqTxSubmitProvider';
+
+const moduleName = 'TxSubmitWorker';
+
+/**
+ * Configuration options parameters for the TxSubmitWorker
+ */
+export interface TxSubmitWorkerConfig {
+  /**
+   * Instructs the worker to process multiple transactions simultaneously.
+   * Default: false (serial mode)
+   */
+  parallel?: boolean;
+
+  /**
+   * The number of Tx submitted to the txSubmitProvider in parallel.
+   * Ignored in serial mode. Default: 3
+   */
+  parallelTx?: number;
+
+  /**
+   * The RabbitMQ polling cycle in milliseconds.
+   * Ignored in parallel mode. Default: 500
+   */
+  pollingCycle?: number;
+
+  /**
+   * The RabbitMQ connection URL
+   */
+  rabbitmqUrl: URL;
+}
+
+/**
+ * Dependencies for the TxSubmitWorker
+ */
+export interface TxSubmitWorkerDependencies {
+  /**
+   * The logger. Default: dummyLogger (no log at all)
+   */
+  logger?: Logger;
+
+  /**
+   * The provider to use to submit tx
+   */
+  txSubmitProvider: TxSubmitProvider;
+}
+
+/**
+ * Controller class for the transactions submission worker which gets
+ * transactions from RabbitMQ and submit them via the TxSubmitProvider
+ */
+export class TxSubmitWorker {
+  /**
+   * The RabbitMQ channel
+   */
+  #channel?: Channel;
+
+  /**
+   * The RabbitMQ connection
+   */
+  #connection?: Connection;
+
+  /**
+   * Flag to exit from the infinite loop in case of sequential tx handlig
+   */
+  #continueForever = false;
+
+  /**
+   * The RabbitMQ consumerTag
+   */
+  #consumerTag?: string;
+
+  /**
+   * Internal messages counter
+   */
+  #counter = 0;
+
+  /**
+   * The function to call to resolve the start method exit Promise
+   */
+  #exitResolver?: () => void;
+
+  /**
+   * The logger
+   */
+  #logger: Logger;
+
+  /**
+   * Worker configuration: it was configured to process multiple tx in parallel
+   */
+  #parallel?: boolean;
+
+  /**
+   * The number of Tx submitted to the txSubmitProvider in parallel.
+   */
+  #parallelTx: number;
+
+  /**
+   * The RabbitMQ polling cycle in milliseconds.
+   */
+  #pollingCycle: number;
+
+  /**
+   * Worker configuration: the RabbitMQ connection URL
+   */
+  #rabbitmqURL: URL;
+
+  /**
+   * The internal worker status
+   */
+  #status: 'connected' | 'connecting' | 'error' | 'idle' = 'idle';
+
+  /**
+   * The TxSubmitProvider instance
+   */
+  #txSubmitProvider: TxSubmitProvider;
+
+  /**
+   * @param {TxSubmitWorkerConfig} options The configuration options
+   */
+  constructor(options: TxSubmitWorkerConfig, dependencies: TxSubmitWorkerDependencies) {
+    const { logger, txSubmitProvider } = dependencies;
+    const { parallel, parallelTx, pollingCycle, rabbitmqUrl } = options;
```

Comment:
The code is more DRY if the whole options object is stored as a class prop. It also then retains the modeling as it's applied throughout other methods

``` ts
this.#options = options
```